lhotse/dataset/surt.py

```diff
@@ -0,0 +1,353 @@
+from collections import defaultdict
+from typing import Callable, Dict, List, Union
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+
+from lhotse import validate
+from lhotse.cut import CutSet
+from lhotse.dataset.input_strategies import BatchIO, PrecomputedFeatures
+from lhotse.utils import LOG_EPSILON, compute_num_frames, ifnone
+from lhotse.workarounds import Hdf5MemoryIssueFix
+
+
+class K2SurtDataset(torch.utils.data.Dataset):
+    """
+    The PyTorch Dataset for the multi-talker ASR task using k2 library.
+    We support the modeling framework known as Streaming Unmixing and Recognition
+    Transducer (SURT), as described in [1] and [2], but this dataset can also be
+    used for other multi-talker ASR approaches such as MT-RNNT [3] and SOT [4].
+    See icefall recipe for usage.
```

Comment:
you might want to link the recipe here -- otherwise it's a nice doc, very helpful :)

Reply:
Yeah, I am cleaning up the icefall recipe to make a PR. I will link it here once it is done.